app/src/main/java/com/example/ithaca_transit_android_v2/NetworkingAdapters.kt

```diff
@@ -44,4 +46,128 @@ class LocationAdapter {
+class RouteAdapter {
+    class DataRoute(
+        val data: DataRouteOptions
+    )
+
+    @JsonClass(generateAdapter = true)
+    class DataRouteOptions(
+        val fromStop: List<JsonRoute>,
+        val walking: List<JsonRoute>,
+        val boardingSoon: List<JsonRoute>
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonRoute(
+        @Json(name = "directions")
+        val directions: List<JsonDirection>,
+        @Json(name = "startCoords")
+        val startLocation: JsonLocation,
+        @Json(name = "endCoords")
+        val endLocation: JsonLocation,
+        // TODO: need to calculate isWalkingOnly
+        @Json(name = "arrivalTime")
+        val arrival: String, // e.g. 2019-11-06T17:07:20Z
+        @Json(name = "departureTime")
+        val depart: String
+        // TODO: need to calculate BoardInMin
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonDirection(
+        @Json(name = "path")
+        val listOfCoordinates: List<Coordinate>,
+        @Json(name = "startTime")
+        val startTime: String,
+        @Json(name = "endTime")
+        val endTime: String,
+        @Json(name = "startLocation")
+        val startLocation: JsonLocation,
+        @Json(name = "endLocation")
+        val endLocation: JsonLocation,
+        @Json(name = "stops")
+        val busStops: List<JsonLocation>,
+        @Json(name = "routeNumber")
+        val busNumber: Int
+    )
+
+    @JsonClass(generateAdapter = true)
+    class JsonLocation(
+        val type: LocationType,
+        val name: String,
+        val lat: Double,
+        val long: Double,
+        val detail: String?
+    )
+
+    @FromJson
+    private fun fromJson(json: DataRoute) : RouteOptions {
+        fun processStringDate(dateString: String): Date {
+            val inputFormatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SS'Z'")
+            return inputFormatter.parse(dateString)
+        }
+
+        fun processStops(locs: List<JsonLocation>): List<Location> {
+            return locs.map{ loc -> Location(LocationType.BUS_STOP, loc.name,
+                Coordinate(loc.lat, loc.long), loc.detail)}
+        }
+
+        fun processDirections(jsonDirection: List<JsonDirection>): List<Direction> {
+            return jsonDirection.map { dir -> Direction(
+                dir.listOfCoordinates.map { c -> Coordinate(c.latitude, c.longitude) },
+                processStringDate(dir.startTime),
+                processStringDate(dir.endTime),
+                Coordinate(dir.startLocation.lat, dir.startLocation.long),
+                Coordinate(dir.endLocation.lat, dir.endLocation.long),
+                processStops(dir.busStops),
+                dir.busNumber
+            )}
+        }
+
+        fun computeBoardInMin(firstBusDirection: Direction): Int {
+            val diff =  firstBusDirection.startTime.time - Calendar.getInstance().time.time
+            return TimeUnit.MINUTES.convert(diff, TimeUnit.MILLISECONDS).toInt()
+        }
```

Comment:
hmm i almost want to move this function into the `Direction` class for reusability

Reply:
Yep! I do not think it belongs in this class and probably be called by Direction.computeBoardInMin.